cms_delete_content/models/website_published.py

```diff
@@ -0,0 +1,46 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017 Camptocamp
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl)
+
+from openerp import api, fields, models, _
+
+
+class WebsitePublishedMixin(models.AbstractModel):
+    _inherit = "website.published.mixin"
+
+    cms_delete_url = fields.Char(
+        string='CMS delete URL',
+        compute='_compute_cms_delete_url',
+        readonly=True,
+    )
+
+    @api.multi
+    def _compute_cms_delete_url(self):
+        for item in self:
+            item.cms_delete_url = \
+                u'/cms/delete/{}/{}'.format(item._name, item.id)
+
+    cms_delete_confirm_url = fields.Char(
+        string='CMS delete confirm URL',
+        compute='_compute_cms_delete_confirm_url',
+        readonly=True,
+    )
+
+    @api.multi
+    def _compute_cms_delete_confirm_url(self):
+        for item in self:
+            item.cms_delete_confirm_url = \
+                u'/cms/delete/{}/{}/confirm'.format(item._name, item.id)
+
+    # customize this per-model
+    cms_after_delete_url = '/'
```

Comment:
This is supposed to be a default for a model. But, you can do conditional stuff in this way:
```
@property
def cms_after_delete_url(self):
    # do your stuff
```
or simpler: use the 'redirect' attribute for "on demand" changes. See https://github.com/simahawk/website-cms/blob/9619b81883f15b33d44582fefa84c38c9582a63a/cms_delete_content/controllers/main.py#L36.